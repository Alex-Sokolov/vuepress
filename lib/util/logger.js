const chalk = require('chalk')

const logger = {}

const logTypes = {
  success: {
    color: 'green',
    label: 'DONE'
  },
  error: {
    color: 'red',
    label: 'FAIL'
  },
  warn: {
    color: 'yellow',
    label: 'WARN'
  },
  tip: {
    color: 'cyan',
    label: 'TIP'
  }
}

const getLoggerFn = (color, label) => (msg, log = true) => {
  msg = chalk.reset.inverse.bold[color](` ${label} `) + ' ' + msg
  return log ? console.log(msg) : msg
}

for (const type in logTypes) {
  const { color, label } = logTypes[type]
  logger[type] = getLoggerFn(color, label)
}

logger.label = (label, msg, color = 'blue', log = true) => getLoggerFn(color, label)(msg, log)

module.exports = logger
