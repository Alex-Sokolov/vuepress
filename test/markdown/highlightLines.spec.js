import { Md, getFragment } from './util'
import highlightLines from '@/markdown/highlightLines.js'

const md = Md()
const mdh = Md().use(highlightLines)

describe('highlightLines', () => {
  test('should the output not change when highlightLines is not detected', async () => {
    const input = await getFragment('code')
    const output1 = md.render(input)
    const output2 = mdh.render(input)
    expect(output1).toBe(output2)
  })

  test('highlight single line', async () => {
    const input = await getFragment('code-highlightLines-single')
    const output = mdh.render(input)
    expect(output).toMatchSnapshot()
  })

  test('highlight multiple lines', async () => {
    const input = await getFragment('code-highlightLines-multiple')
    const output = mdh.render(input)
    expect(output).toMatchSnapshot()
  })
})
