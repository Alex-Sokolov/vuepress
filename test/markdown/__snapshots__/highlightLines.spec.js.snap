// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlightLines highlight multiple lines 1`] = `
<div class="highlight-lines">
  <div class="highlighted">&nbsp;</div>
  <div class="highlighted">&nbsp;</div>
  <br>
  <div class="highlighted">&nbsp;</div>
  <div class="highlighted">&nbsp;</div>
  <br>
  <br>
</div><pre v-pre class="language-js"><code><span class="token keyword">const</span> app <span class="token operator">=</span> <span class="token keyword">new</span> <span class="token class-name">Vue</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  render<span class="token punctuation">,</span>
  router
<span class="token punctuation">}</span><span class="token punctuation">)</span>

app<span class="token punctuation">.</span><span class="token function">$mount</span><span class="token punctuation">(</span><span class="token string">'#app'</span><span class="token punctuation">)</span>
</code></pre>
`;

exports[`highlightLines highlight single line 1`] = `
<div class="highlight-lines">
  <div class="highlighted">&nbsp;</div>
  <br>
</div><pre v-pre class="language-js"><code><span class="token keyword">new</span> <span class="token class-name">Vue</span><span class="token punctuation">(</span><span class="token punctuation">)</span>
</code></pre>
`;
